staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -347,6 +366,10 @@ func (w *watchCache) doCacheResizeLocked(capacity int) {
 	if capacity < w.capacity {
 		// adjust startIndex if cache capacity shrink.
 		w.startIndex = w.endIndex - capacity
+		if w.snapshots != nil {
+			oldestRV := w.cache[w.startIndex%w.capacity].ResourceVersion	:
+			w.snapshots.RemoveLessOrEqual(oldestRV)
```

Comment:
This seems like off-by-one error here - the oldestRV used here is an RV that we still technically support.
So it seems that "OrEqual" part removes something that we need.